js/oop/Palindrome.js

```diff
@@ -12,6 +12,49 @@ export class Palindrome {
   }
 
   // write your code here
+  reverse() {
+    const cleanedValue = this.value.toLowerCase().replace(/[^a-z0-9]/g, "");
+
+    const reversedValue = cleanedValue.split("").reverse().join("");
+
+    return cleanedValue === reversedValue;
+  }
+
+  loop() {
+    const cleanedValue = this.value.toLowerCase().replace(/[^a-z0-9]/g, "");
+
+    const length = cleanedValue.length;
+
+    for (let i = 0; i < length / 2; i++) {
+      if (cleanedValue[i] !== cleanedValue[length - 1 - i]) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  recursive() {
+    this.value = this.value.replace(/[^a-zA-Z]/g, "").toLowerCase();
+  }
+
+  isPalindromeRecursive(index = 0) {
+    // Clean the string before checking for palindrome
+    this.recursive();
+
+    // Base case: if the index reaches the middle of the string or beyond, the string is a palindrome
+    if (index >= Math.floor(this.value.length / 2)) {
+      return true;
+    }
+
+    // If the character at the first index and the last index are not the same, the string is not a palindrome
+    if (this.value[index] !== this.value[this.value.length - 1 - index]) {
+      return false;
+    }
+
+    // Recursive call to check the next pair of characters towards the middle
+    return this.isPalindromeRecursive(index + 1);
+  }
```

Comment:
Can you explain why you have to make these two methods recursive() and isPalindromeRecursive() while these methods likely do very different ways?

Reply:
I made a mistake in naming the method, the first one should be to cleaned the value.